test/unit/BNum.t.sol

```diff
@@ -3,379 +3,342 @@ pragma solidity 0.8.25;
+    vm.expectRevert(BNum.BNum_DivZero.selector);
 
-    // Action
-    bNum.bdivExposed(1, 0);
+    bNum.call_bdiv(1, 0);
   }
 
   function test_BdivRevertWhen_PassingAAsUint256Max() external {
-    // Post Condition
     // it should revert
-    vm.expectRevert('ERR_DIV_INTERNAL');
+    vm.expectRevert(BNum.BNum_DivInternal.selector);
 
-    // Action
-    bNum.bdivExposed(type(uint256).max, 1);
+    bNum.call_bdiv(type(uint256).max, 1);
   }
 
   function test_BdivWhenPassingAAndBTooBig() external {
-    // Preconditions
     uint256 _a = type(uint256).max;
     uint256 _b = BONE;
 
-    // Post Condition
     // it should revert
     //     a*BONE/b + b/2 > uint256 max
-    vm.expectRevert('ERR_DIV_INTERNAL');
+    vm.expectRevert(BNum.BNum_DivInternal.selector);
 
-    // Action
-    bNum.bdivExposed(_a, _b);
+    bNum.call_bdiv(_a, _b);
+  }
+
+  function test_BdivWhenFlooringToZero() external {
+    //     (1 * BONE) / 2 * BONE + 1 = 0.499..
+    uint256 _a = 1;
+    uint256 _b = 2e18 + 1;
+
+    uint256 _result = bNum.call_bdiv(_a, _b);
+
+    // it should return 0
+    assertEq(_result, 0);
   }
 
   function test_BDivWhenFlooringToZero(uint256 _a, uint256 _b) external {
-    // Preconditions
     _a = bound(_a, 1, (type(uint256).max / (BONE * 2)) - 1);
     _b = bound(_b, (2 * BONE * _a) + 1, type(uint256).max);
 
-    // Action
-    uint256 _result = bNum.bdivExposed(_a, _b);
+    uint256 _result = bNum.call_bdiv(_a, _b);
 
-    // Post Condition
     // it should return 0
     assertEq(_result, 0);
   }
 
   function test_BdivWhenPassingKnownValues() external {
     // it should return correct value
-    vm.skip(true);
+    //     5 / 2 = 2.5
+    uint256 _a = 5e18;
+    uint256 _b = 2e18;
+
+    uint256 _result = bNum.call_bdiv(_a, _b);
+
+    assertEq(_result, 2.5e18);
   }
 
-  function test_BpowiWhenPassingAAsZero() external {
-    // Action
-    uint256 _result = bNum.bpowiExposed(0, 3);
+  function test_BpowiWhenPassingBaseAsZero() external {
+    uint256 _result = bNum.call_bpowi(0, 3);
 
-    // Post Condition
     // it should return zero
     assertEq(_result, 0);
   }
 
-  function test_BpowiWhenPassingBAsZero() external {
-    // Action
-    uint256 _result = bNum.bpowiExposed(3 * BONE, 0);
+  function test_BpowiWhenPassingExponentAsZero() external {
+    uint256 _result = bNum.call_bpowi(3e18, 0);
 
-    // Post Condition
     // it should return BONE
     assertEq(_result, BONE);
   }
 
-  function test_BpowiWhenPassingAAsOne() external {
-    // Action
-    uint256 _result = bNum.bpowiExposed(BONE, 3);
+  function test_BpowiWhenPassingBaseAsOne() external {
+    uint256 _result = bNum.call_bpowi(BONE, 3);
 
-    // Post Condition
     // it should return BONE
     assertEq(_result, BONE);
   }
 
   function test_BpowiWhenPassingKnownValues() external {
     // it should return correct value
-    vm.skip(true);
+    //     4. ^ 12 = 16777216
+    uint256 _a = 4e18;
+    uint256 _b = 12;
+
```

Comment:
nit: for small & simple cases I'd skip the whitespace separating actions and {pre,post}conditions

Reply:
doesn't hurt tho 😇